src/shared/lib/fetch/index.js

```diff
@@ -162,17 +162,29 @@ export const pdf = async (scraper, url, cacheKey = 'default', date, options) =>
   return data;
 };
 
-const fetchHeadless = async url => {
+/**
```

Comment:
This is very confusing, why are you mixing async primitives (async/await, callback)? What is the intended outcome?

Reply:
This looks ok to me.  The headless browser is kept within the fetch API, and some actions need to be performed on the page once it has loaded.  Passing those actions to the fetch makes more sense than returning the full puppeteer `page` object back to the caller, and is what we would have to do in Li.

Is it the term `callback` that is confusing?

Reply:
Yep, that’s exactly what my intention was. 

I didn’t want to have the page leak out of this function so that within this function we can ensure thy the browser is closed and resources are cleaned up. 

Callback might be the wrong name here

Reply:
Cheers, taking a look.